zerver/views/streams.py

```diff
@@ -532,6 +526,9 @@ def add_subscriptions_backend(
         user_profile, existing_streams
     )
     if len(unauthorized_streams) > 0 and authorization_errors_fatal:
+        if user_profile.is_guest:
+            raise JsonableError(_("Not allowed for guest users"))
```

Comment:
I am not sure what error message would be best here and below after `if len(principals) > 0:`, but kept it same as before for now.